lib/hammer_cli_foreman/filter.rb

```diff
@@ -43,17 +43,18 @@ def extend_data(filter)
 
 
     module TaxonomyCheck
-      def taxonomy_options?
-        option_location_names || option_location_ids || option_organization_names || option_organization_ids
-      end
-
-      def signal_override_usage_error
-        signal_usage_error _('Organizations and locations can be set only for overriding filters')
-      end
 
       def self.included(base)
+        def taxonomy_options?
+          option_location_names || option_location_ids || option_organization_names || option_organization_ids
```

Comment:
Maybe it would be more intuitive to run the validators after the option_source evaluation. There we could just test `organization_id` instead of the list of all the possible names.  Or add `ref(resource)` that would expand according to searchables defined for the resource.

Reply:
I'd vote rather for B. Switching the order of evaluation can easily break validations created to avoid some unexpected id resolution scenarios (e.g. in katello?).

It might also be useful to switch from one validation block to validation objects that would be part of the option source pipeline. That would allow easier composition and enable validation at any stage. But this is definitely for future refactoring, not part of this PR.

Reply:
I fixed the condition and created RFE [1] For the validation object I like the idea.
[1] http://projects.theforeman.org/issues/22253